src/test/spec/initial_dns_seedlist_discovery.rs

```diff
@@ -23,162 +24,194 @@ struct TestFile {
 }
 
 #[derive(Debug, Deserialize, PartialEq)]
-#[serde(rename_all = "camelCase")]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
 struct ResolvedOptions {
     replica_set: Option<String>,
     auth_source: Option<String>,
     ssl: bool,
+    load_balanced: Option<bool>,
+    direct_connection: Option<bool>,
+}
+
+impl ResolvedOptions {
+    fn assert_eq(&self, options: &ClientOptions) {
+        // When an `authSource` is provided without any other authentication information, we do
+        // not keep track of it within a Credential. The options present in the spec tests
+        // expect the `authSource` be present regardless of whether a Credential should be
+        // created, so the value of the `authSource` is not asserted on to avoid this
+        // discrepancy.
+        assert_eq!(self.replica_set, options.repl_set_name);
+        assert_eq!(self.ssl, options.tls_options().is_some());
+        assert_eq!(self.load_balanced, options.load_balanced);
+        assert_eq!(self.direct_connection, options.direct_connection);
+    }
 }
 
 #[derive(Debug, Deserialize, Default, PartialEq)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
 struct ParsedOptions {
     user: Option<String>,
     password: Option<String>,
     db: Option<String>,
 }
 
-#[cfg_attr(feature = "tokio-runtime", tokio::test)]
-#[cfg_attr(feature = "async-std-runtime", async_std::test)]
-async fn run() {
-    async fn run_test(mut test_file: TestFile) {
-        // TODO DRIVERS-796: unskip this test
-        if test_file.uri == "mongodb+srv://test5.test.build.10gen.cc/?authSource=otherDB" {
-            return;
-        }
-
-        // "encoded-userinfo-and-db.json" specifies a database name with a question mark which is
-        // disallowed on Windows. See
-        // <https://docs.mongodb.com/manual/reference/limits/#restrictions-on-db-names>
-        if let Some(ref mut options) = test_file.parsed_options {
-            if options.db.as_deref() == Some("mydb?") && cfg!(target_os = "windows") {
-                options.db = Some("mydb".to_string());
-                test_file.uri = test_file.uri.replace("%3F", "");
-            }
-        }
+async fn run_test(mut test_file: TestFile) {
+    // TODO DRIVERS-796: unskip this test
+    if test_file.uri == "mongodb+srv://test5.test.build.10gen.cc/?authSource=otherDB" {
+        return;
+    }
 
-        let result = if cfg!(target_os = "windows") {
-            ClientOptions::parse_with_resolver_config(&test_file.uri, ResolverConfig::cloudflare())
-                .await
-        } else {
-            ClientOptions::parse(&test_file.uri).await
-        };
+    // TODO RUST-979 unskip these tests
```

Comment:
This will require an evergreen config for running tests against a loadbalancer, which will come in a follow-up PR.